File: app/src/main/java/com/example/starwars/viewmodel/FilmListActivityViewModel.java
package com.example.starwars.viewmodel;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;

import com.example.starwars.model.Film;
import com.example.starwars.repository.FilmsRepository;

import java.util.List;

public class FilmListActivityViewModel extends ViewModel {

    private LiveData<List<Film>> mFilms;
    private FilmsRepository mRepository;
    private LiveData<Boolean> mIndicator;

    public void init() {
        mRepository = FilmsRepository.getInstance();
        mIndicator = mRepository.getLoadingIndicator();
        loadData();
    }

    public LiveData<List<Film>> getFilms() {
        return mFilms;
    }
    public LiveData<Boolean> getLoadingIndicator() {
        return mIndicator;
    }
    public void loadData() {
        mFilms = mRepository.getDataFromAPI();
    }

}

File: app/src/main/java/com/example/starwars/viewmodel/FilmListActivityViewModel.kt
package com.example.starwars.viewmodel

import android.util.Log
import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModel
import com.example.starwars.activity.FilmListActivity.Companion.TAG
import com.example.starwars.model.Film
import com.example.starwars.repository.FilmsRepository
import dagger.hilt.android.lifecycle.HiltViewModel
import javax.inject.Inject

@HiltViewModel
class FilmListActivityViewModel @Inject constructor(private var mRepository: FilmsRepository) : ViewModel() {
    var currentPage = 1
    lateinit var filmsLiveData: LiveData<List<Film>?>
        private set
    lateinit var indicatorLiveData: LiveData<Boolean>
        private set

    fun init() {
        indicatorLiveData = mRepository.loadingIndicator
    }

    fun loadData(isInternetConnected: Boolean) {
        Log.d(TAG, "init: after loadDataCalled")
        if (isInternetConnected) filmsLiveData = mRepository.filmListFromAPI(currentPage)
        else filmsLiveData = mRepository.getFilmListFromDatabase()
    }
}